### ISSUE TITLE
Feature request: non-blocking scroll

### ISSUE THREAD

**Issue by francisdb** [sksamuel/elastic4s#1141]
Preferably using the reactive streams low-level api so users can bridge to any other streaming framework.

From the README
> The SearchIterator will do this for you, although it will block between requests.

**Comment by sksamuel** [comment]
What would you like to see on top of the standard scroll API?

**Comment by francisdb** [comment]
For a rationale see http://www.reactive-streams.org/#the-problem

These blocking calls would disappear with a non-blocking api
https://github.com/sksamuel/elastic4s/blob/master/elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/search/SearchIterator.scala#L43

**Comment by sksamuel** [comment]
Right, but what I really mean is, what's wrong with the existing scroll API which is non-blocking or the existing elastic4s-streams impl which is non blocking. Is there something extra you'd like to see on top?

**Comment by francisdb** [comment]
I see, I missed that `elastic4s-streams` extension, sorry about that.

**Comment by sksamuel** [comment]
No problem :)

**Comment by francisdb** [comment]
@sksamuel I see the `elastic4s-http-streams` is not available for 6.0?

**Comment by sksamuel** [comment]
6.0 final isn't released yet, but it's only an hour away, and that will include http streams.
